### PR TITLE
gitserver: make `ReposStats` use `gitserverClient.do` for sending HTTP requests.

### PR DESCRIPTION
**Pull request by sashaostrikov**
This commit makes sending HTTP requests uniform across the gitserver client, using `gitserverClient.do` which includes all necessary headers.

Repo name which is used in `do` only for tracing purpose is passed as `repos-stats call, no repo provided` because `ReposStats` doesn't use repos, it uses gitserver instance address.

Test plan:
New unit test added.

Closes https://github.com/sourcegraph/sourcegraph/issues/43221

### CODE REVIEW

File: internal/gitserver/client.go [hunk 1]
```diff
@@ -1143,23 +1143,7 @@ func (c *clientImplementor) ReposStats(ctx context.Context) (map[string]*protoco
...
-	// We should consolidate into a single method for creating requests so that we are
-	// setting all the required properties like headers, trace spans in a central place.
-	req.Header.Set("Content-Type", "application/json")
-	req.Header.Set("User-Agent", c.userAgent)
-
-	// Set header so that the server knows the request is from us.
-	req.Header.Set("X-Requested-With", "Sourcegraph")
-
-	resp, err := c.httpClient.Do(req)
+	resp, err := c.do(ctx, "repos-stats call, no repo provided", "GET", fmt.Sprintf("http://%s/repos-stats", addr), nil)
```
Comment (ryanslade) [inline internal/gitserver/client.go:1146]: Passing in the weird repo name is.. weird.

Maybe we can have two versions of `do`? One that takes a repo and another that doesn't? 

The one that that doesn't can just pass an empty string to the other.

Also, see the comment above "do"? Maybe we can remove the "GET" param as it's always the same

Comment (mrnugget) [reply internal/gitserver/client.go:1146]: Agree. The fake repo-name is more confusing than the previously duplicated code, IMHO

Comment (sashaostrikov) [reply internal/gitserver/client.go:1146]: I began with an empty string (repo name), but then changed it. I can either pass an empty string to trace attribute, or just in this case remove this attribute altogether.

About "GET": I'll remove this input argument 👍🏻

WDYT? @ryanslade @mrnugget

Comment (ryanslade) [reply internal/gitserver/client.go:1146]: I think if the string is missing then don't add it as an attribute

**Comment by sashaostrikov** [comment]
@mrnugget @ryanslade 
In fact, now both `POST` and `GET` methods are used, that's why I just dropped the suppress warnings statement from `do` function doc.

File: internal/gitserver/client.go [hunk 2]
```diff
@@ -1238,9 +1222,8 @@ func (c *clientImplementor) httpPostWithURI(ctx context.Context, repo api.RepoNa
 	return c.do(ctx, repo, "POST", uri, b)
 }
 
-// do performs a request to a gitserver instance based on the address in the uri argument.
-//
-//nolint:unparam // unparam complains that `method` always has same value across call-sites, but that's OK
+// do performs a request to a gitserver instance based on the address in the uri
+// argument.
```
Comment (mrnugget) [inline internal/gitserver/client.go:1226]: Can you also add something here to say that the `repo` is optional?

**Review by mrnugget — APPROVED** [review]
Nice. Did you test this locally by running the environment and making sure this is called?

**Comment by sashaostrikov** [comment]
@mrnugget yep, the number which is calculated by this function is rendered locally
image

**Review by ryanslade — COMMENTED** [review]
Nice!

**Pull request merged**